### PR TITLE
feat: edit contact from address book

### PR DESCRIPTION
**Pull request by patricio0312rev**
# [[address book] handle edit](https://app.clickup.com/t/86dtbmvhc)

## Summary

- Depends on #294
- `Add Contact` page's components have been refactored.
- Types from the address book section have been moved to their own file.
- New `Edit Contact` page.

https://github.com/ArdentHQ/arkconnect-extension/assets/55117912/413fe45b-98ea-4c4d-9a35-a4a0e3ed49f8

## Checklist

- [ ] I checked that both `pnpm dev` and `pnpm dev:bare` work as intended
- [ ] I checked the basic extension interactions and made sure wallet selection works
- [ ] I checked my UI changes against the design and there are no notable differences, including responsiveness
- [ ] I checked my (code) changes for obvious issues, debug statements and commented code
- [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
- [ ] I added a short description on how to test this PR _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: src/pages/EditContact.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { useNavigate, useParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { useQuery } from 'react-query';
+import { object, string } from 'yup';
+import { useFormik } from 'formik';
+import { useEffect, useState } from 'react';
+import { fetchValidateAddress } from './CreateContact';
```
Comment (ItsANameToo) [inline src/pages/EditContact.tsx:7]: added a card to refactor this: https://app.clickup.com/t/86dth2jgd

File: src/pages/EditContact.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        name: string()
+            .required(t('ERROR.IS_REQUIRED', { name: 'Name' }))
+            .max(20, t('ERROR.MAX_CHARACTERS', { count: 20 }))
+            .test('unique-name', t('ERROR.IS_DUPLICATED', { name: 'contact name' }), (name) => {
+                if (contact?.name === name) return true;
+                return !addressBook?.find((contact) => contact.name === name);
+            }),
+        address: string()
+            .required(t('ERROR.IS_REQUIRED', { name: 'Address' }))
+            .min(34, t('ERROR.IS_INVALID', { name: 'Address' }))
```
Comment (ItsANameToo) [inline src/pages/EditContact.tsx:43]: 34 min but also max

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

**Comment by ItsANameToo** [comment]
conflict @patricio0312rev

**Review by samharperpittam — CHANGES_REQUESTED** [review]
When running `pnpm dev` and attempting to edit one of the seeded contacts, the edit functionality doesn't work (the form doesn't open). Is it possible to fix that?

However, editing a new contact that gets added works fine.

**Review by samharperpittam — CHANGES_REQUESTED** [review]
When editing a contact within the address book, if you modify the name of the contact, the edited name gets added as the address.

See video here: https://helloardent.slack.com/archives/C069X9DHWUR/p1715694763021289

**Comment by ItsANameToo** [comment]
some conflicts @patricio0312rev

**Pull request merged**